pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -591,6 +591,9 @@ func newNSTemplateSet(userAcc *toolchainv1alpha1.UserAccount) *toolchainv1alpha1
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      userAcc.Name,
 			Namespace: userAcc.Namespace,
+			Labels: map[string]string{
```

Comment:
As I mentioned before, the provider label is for objects created by the NSTempalateSet controller only. Not for the NSTemplateSet itself.